ui/app/src/components/BluePrintForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect } from 'react';
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import Grid from '@material-ui/core/Grid';
+import FormHelperText from "@material-ui/core/FormHelperText";
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import Input from '@material-ui/core/Input';
+import TextField from '@material-ui/core/TextField';
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import Collapse from '@material-ui/core/Collapse';
+import GitForm from "./GitForm";
+
+const useStyles = makeStyles(theme => ({
+  form: {
+    maxWidth: '600px',
+    margin: 'auto',
+    paddingBottom: '20px'
+  }
+}));
+
+function BluePrintForm(props: any) {
```

Comment:
Let’s define the props interface for the component. Do it once you’re sure about the props, though.